### PR TITLE
functions --hooks

### PR DESCRIPTION
**Pull request by hamon-e**
Add a little options to list all events hooks
The current output it's not really pretty but i had no idea to make it better

It's my first contribution don't hesitate to criticize

### CODE REVIEW

**Comment by ridiculousfish** [comment]
This looks nice! Thanks for doing this!

I don' t know about the name "hooks," we don't use that term anywhere else. Users cause a  function to respond to an event via `--on-event`, how about "event handlers?"

A few nits I'll put in code review.

**Comment by ridiculousfish** [comment]
This looks nice! Thanks for doing this!

I don' t know about the name "hooks," we don't use that term anywhere else. Users cause a  function to respond to an event via `--on-event`, how do you feel about "event handlers?"

A few nits I'll put in code review. Also if you get to it you can add this to the CHANGELOG, if not I'll do it.

File: src/builtin_functions.cpp [hunk 1]
```diff
@@ -35,15 +35,18 @@ struct functions_cmd_opts_t {
...
 };
-static const wchar_t *short_options = L":Dacehnqv";
+static const wchar_t *short_options = L":oDacehnqv";
 static const struct woption long_options[] = {
     {L"erase", no_argument, NULL, 'e'},   {L"description", required_argument, NULL, 'd'},
     {L"names", no_argument, NULL, 'n'},   {L"all", no_argument, NULL, 'a'},
     {L"help", no_argument, NULL, 'h'},    {L"query", no_argument, NULL, 'q'},
     {L"copy", no_argument, NULL, 'c'},    {L"details", no_argument, NULL, 'D'},
-    {L"verbose", no_argument, NULL, 'v'}, {NULL, 0, NULL, 0}};
+    {L"verbose", no_argument, NULL, 'v'}, {L"hooks", optional_argument, NULL, 'o'},
```
Comment (jchilders) [inline src/builtin_functions.cpp]: Seconding @ridiculousfish's comment: "event_handlers" would be more consistent.

**Comment by hamon-e** [comment]
yeah i prefer it too ; what do you think of it ?

**Comment by hamon-e** [comment]
I don't find optional argument really pratical
so i create another one

File: src/event.cpp [hunk 2]
```diff
@@ -448,6 +448,67 @@ void event_fire(const event_t *event) {
     }
 }
 
+void event_print(io_streams_t &streams, const wcstring *filter) {
+    std::vector<shared_ptr<event_t>> tmp;
+
+    static std::map<int, wcstring> dico = {
```
[the remark this answers is not in the document]
Comment (hamon-e) [reply src/event.cpp]: I also need to do the reverse operation (have a string and get a int)
and for that i can't do a switch statement

**Comment by faho** [comment]
What's the status here? @ridiculousfish: Did you forget about this?

**Comment by ridiculousfish** [comment]
It's been in my dock all this time :/

**Comment by ridiculousfish** [comment]
Merged as 3819437e0edcf21e770456856adbee0bfec875a2, documented as 9a5afe3913a66ff5353898327b86ee0efdd68326.  Thanks!

**Pull request closed without merging**